src/js/hooks/useNavigation.jsx

```diff
@@ -0,0 +1,107 @@
+const useNavigation = ({
+  newRowFocusColumns,
+  existingRowFocusColumns,
+  tableData,
+  setFocusId,
+  setFocusIndex,
+  addEmptyRow,
+  productCode,
+  cycleCountId,
+}) => {
+  const getNextFocus = (columnId, rowIndex) => {
+    const currentIndex = newRowFocusColumns.indexOf(columnId);
+    const remainingColumns = newRowFocusColumns.slice(currentIndex + 1);
+    const newRowIndex = tableData[rowIndex + 1];
+
+    if (!remainingColumns.some(col => existingRowFocusColumns.includes(col))) {
+      if (!newRowIndex) {
+        addEmptyRow(productCode, cycleCountId);
```

Comment:
It shouldn't be named `cycleCountId` because this hook should be generic, but this name is feature-specific.

Reply:
@SebastianLib this should be generic not only for CycleCount.